internal/evalchecks/eval_for_each.go

```diff
@@ -2,33 +2,40 @@
 // SPDX-License-Identifier: MPL-2.0
 // Copyright (c) 2023 HashiCorp, Inc.
 // SPDX-License-Identifier: MPL-2.0
+package evalchecks
 
-package tofu
+// TODO this is only here until i figure out where it should be
```

Comment:
Is it still relevant?

Reply:
kind of. I still feel like there is maybe somewhere better it should be? but no-one seems to mind it being here...

Reply:
I suggested maybe in the lang package.  If it does not fit there, I think leaving it as it's own package works well enough for now.

Reply:
import cycled in the lang package itself. but I moved it into a sub package and it felt like it might be more at home there?

Reply:
Sounds good to me!